src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Reindex/ReindexHandlerTests.cs

```diff
@@ -87,11 +87,12 @@ public async Task GivenAGetRequest_WhenTooManyRequestsThrown_ThenTooManyRequests
 
             var jobRecord = new ReindexJobRecord(_resourceTypeSearchParameterHashMap, 1);
             var jobWrapper = new ReindexJobWrapper(jobRecord, WeakETag.FromVersionId("id"));
-            _fhirOperationDataStore.GetReindexJobByIdAsync("id", CancellationToken.None).Throws(new RequestRateExceededException(TimeSpan.FromMilliseconds(100)));
+            _fhirOperationDataStore.GetReindexJobByIdAsync("id", CancellationToken.None).Throws(new Exception(null, new RequestRateExceededException(TimeSpan.FromMilliseconds(100))));
```

Comment:
>new Exception(null [](start = 96, length = 18)

I find this route is a bit counterintuitive.
Why can't we peel cosmos exception and pop to the top it's inner exception rather than bend rest of code for it?

Reply:
The issue is that there are many entrypoints to the cosmos SDK (basically the methods on `Container`), and therefore many places where we would have to make sure the exception is mapped properly. Prior to adopting the v3 SDK, we used to have a generated wrapper to the SDK that could forward calls to the and would do common pre/post processing. We could do something similar again.

Reply:
Ok, it's probably fine as long as it's not leaking into non cosmos specific code.